SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/SneakerEntryListActivity.java

```diff
@@ -0,0 +1,176 @@
+		Sneaker selectedSneaker = mSneakers.get(position);
+		Intent i = new Intent(SneakerEntryListActivity.this,
+				SneakerDetailsActivity.class);
+		i.putExtra(LIST_NAME_EXTRA, selectedSneaker.getName());
+		i.putExtra(LIST_BRAND_EXTRA, selectedSneaker.getBrand());
+		i.putExtra(LIST_SELLING_EXTRA, selectedSneaker.getSellingValue());
+		i.putExtra(LIST_RARITY_EXTRA, selectedSneaker.getRarity());
+		i.putExtra(LIST_CATEGORY_EXTRA, selectedSneaker.getCategory().getName());
+		i.putExtra(LIST_DESC_EXTRA, selectedSneaker.getDescription());
+		i.putExtra(LIST_IMGID_EXTRA, selectedSneaker.getThumbnailId());
+		i.putExtra(LIST_TITLE_EXTRA, selectedSneaker.getTitleName());
+		i.putExtra(LIST_ID_EXTRA, selectedSneaker.getId().toString());
+		startActivityForResult(i, 1);
+
+	}
+
+	@Override
+	public boolean onCreateOptionsMenu(Menu menu) {
+		menu.add("Add Sneak Peek Entry");
+		return super.onCreateOptionsMenu(menu);
+	}
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item) {
+		if (item.getTitle() == "Add Sneak Peek Entry") {
+			Intent i = new Intent(SneakerEntryListActivity.this,
+					EditSneakerEntryActivity.class);
+			i.putExtra(LIST_CATEGORY_EDIT_EXTRA, this.mCategory.getName()
+					.toString());
+			startActivityForResult(i, 1);
+		}
+		return super.onOptionsItemSelected(item);
+	}
+
+	@Override
+	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+		updateList();
+	}
+
+	private void updateList() {
+		ArrayList<Sneaker> sneakers = SneakerDirectory.get(
+				getApplicationContext()).getSneakersByCategory(
+				mCategory.getName());
+		sortEntriesAlphabetically(sneakers);
+		mAdapter.clear();
+		mAdapter.addAll(sneakers);
+		mAdapter.notifyDataSetChanged();
+	}
+
+	@Override
+	public void onCreateContextMenu(ContextMenu menu, View v,
+			ContextMenuInfo menuInfo) {
+		super.onCreateContextMenu(menu, v, menuInfo);
+		MenuInflater inflater = getMenuInflater();
+		inflater.inflate(R.menu.sneaker_list_context_menu, menu);
+	}
+
+	@Override
+	public boolean onContextItemSelected(MenuItem item) {
+		AdapterContextMenuInfo info = (AdapterContextMenuInfo) item
+				.getMenuInfo();
+		int position = info.position;
+		Sneaker sneaker = mAdapter.getItem(position);
+
+		switch (item.getItemId()) {
+		case R.id.menu_list_edit_sneaker:
+			Intent i = new Intent(SneakerEntryListActivity.this,
+					EditSneakerEntryActivity.class);
+			i.putExtra(LIST_NAME_EXTRA, sneaker.getName());
+			i.putExtra(LIST_BRAND_EXTRA, sneaker.getBrand());
+			i.putExtra(LIST_SELLING_EXTRA, sneaker.getSellingValue());
+			i.putExtra(LIST_RARITY_EXTRA, sneaker.getRarity());
+			i.putExtra(LIST_CATEGORY_EXTRA, sneaker.getCategory().getName()
+					.toString());
+			i.putExtra(LIST_DESC_EXTRA, sneaker.getDescription());
+			i.putExtra(LIST_IMGID_EXTRA, sneaker.getThumbnailId());
+			i.putExtra(LIST_TITLE_EXTRA, sneaker.getTitleName());
+			i.putExtra(LIST_ID_EXTRA, sneaker.getId().toString());
+			i.putExtra(LIST_FROMLIST_EXTRA, true);
```

Comment:
Implement Sneaker model as Serializable and replace this with the sneaker object itself